File: generator/source/mavlink/src/main/scala/ru/mipt/acsl/decode/mavlink/generator/MavlinkSourceGeneratorConfig.scala
package ru.mipt.acsl.decode.mavlink.generator

/**
  * Created by metadeus on 21.05.16.
  */
trait MavlinkSourceGeneratorConfig {

  def includeContents(fileName:  String): String

  def inputContents: String

  def writeOutput(contents: String): Unit

  def nsFqn: String

  def componentName: String

}

File: generator/source/mavlink/src/main/java/ru/mipt/acsl/decode/mavlink/generator/MavlinkSourceGeneratorConfig.java
package ru.mipt.acsl.decode.mavlink.generator;

/**
 * Created by metadeus on 08.06.16.
 */
public interface MavlinkSourceGeneratorConfig {

    String getIncludeContents(String fileName);

    String getInputContents();

    void writeOutput(String contents);

    String getNamespaceFqn();

    String getComponentName();

}
